Testscripts/Windows/DM-MINMEMHONOR.ps1

```diff
@@ -55,89 +55,97 @@ param([String] $TestParams,
     $resultArr = @()
     try {
         $testResult = $null
-        $vm1name = $VM1.RoleName
-        $vm2name = $VM2.RoleName
-        $HvServer = $VM1.HyperVHost
-        $VM1Ipv4 = $VM1.PublicIP
-        $VM2Ipv4 = $VM2.PublicIP
-        $VMPort = $VM2.SSHPort
+        $vm1name = $AllVmData[0].RoleName
+        $vm2name = $AllVmData[1].RoleName
+        $HvServer = $AllVmData[0].HyperVHost
+        $VM1Ipv4 = $AllVmData[0].PublicIP
+        $VM2Ipv4 = $AllVmData[1].PublicIP
+        $VMPort = $AllVmData[0].SSHPort
+
+        # change working directory to root dir
+        Set-Location $WorkingDirectory
+
         Write-LogInfo "VM1name is $vm1name"
         Write-LogInfo "VM2name is $vm2name"
         Write-LogInfo "Hvserver is $HvServer"
-        Write-LogInfo "Param vm1 Details are -VM $VM1 -minMem $TestParams.minMem -maxMem $TestParams.maxMem -startupMem $TestParams.startupMem -memWeight $TestParams.memWeight"
-        Write-LogInfo "Param vm2 Details are -VM $VM2 -minMem $TestParams.minMem1 -maxMem $TestParams.maxMem1 -startupMem $TestParams.startupMem1 -memWeight $TestParams.memWeight1"
-        Set-VMDynamicMemory -VM $VM1 -minMem $TestParams.minMem -maxMem $TestParams.maxMem `
+
+        Write-LogInfo "Param vm1 Details are -VM $vm1name -minMem $($TestParams.minMem) -maxMem $($TestParams.maxMem) -startupMem $($TestParams.startupMem) -memWeight $($TestParams.memWeight)"
+        Write-LogInfo "Param vm2 Details are -VM $vm2name -minMem $($TestParams.minMem1) -maxMem $($TestParams.maxMem1) -startupMem $($TestParams.startupMem1) -memWeight $($TestParams.memWeight1)"
+        Set-VMDynamicMemory -VM $AllVmData[0] -minMem $TestParams.minMem -maxMem $TestParams.maxMem `
             -startupMem $TestParams.startupMem -memWeight $TestParams.memWeight | Out-Null
-        Set-VMDynamicMemory -VM $VM2 -minMem $TestParams.minMem1 -maxMem $TestParams.maxMem1 `
+        Set-VMDynamicMemory -VM $AllVmData[1] -minMem $TestParams.minMem1 -maxMem $TestParams.maxMem1 `
             -startupMem $TestParams.startupMem1 -memWeight $TestParams.memWeight1 | Out-Null
-        Write-LogInfo "Starting VM1 $vm1name"
-        $VM1Ipv4 = Start-VMandGetIP $vm1name $HvServer $VMPort $user $password
-        Write-LogInfo "IP address of the VM $vm1name is  $VM1Ipv4"
-        # change working directory to root dir
-        Set-Location $WorkingDirectory
+
         $vm1 = Get-VM -Name $vm1name -ComputerName $HvServer -ErrorAction SilentlyContinue
         $vm2 = Get-VM -Name $vm2name -ComputerName $HvServer -ErrorAction SilentlyContinue
-        Write-LogInfo "Starting VM2 $vm2name"
-        $VM2Ipv4 = Start-VMandGetIP $vm2name $HvServer $VMPort $user $password
-        Write-LogInfo "IP address of the VM $vm2name is  $VM2Ipv4"
+        Write-LogInfo "Starting VM1 $vm1name"
+        $VM1Ipv4 = Start-VMandGetIP $vm1name $HvServer $VMPort $user $password
+        Write-LogInfo "IP address of the VM $vm1name is $VM1Ipv4"
+
+        for($i=1; $i -le 10; $i++) {
```

Comment:
This logic should be revised. the first VM powered one before, and here the second VM powered on and check the IP address? Propose two VM powered on before, and save the time from getting an IP address during run-time

Reply:
https://github.com/LIS/LISAv2/pull/673/files#diff-6aaf12de08bc8c767edc25df682495bcR82 those setting will shutdown - start VM, the IP may change.